src/include/miopen/convolution.hpp

```diff
@@ -33,12 +33,16 @@
 #include <miopen/solver_id.hpp>
 #include <miopen/names.hpp>
 
+#include <boost/any.hpp>
+
 #include <string>
 #include <tuple>
 #include <vector>
 
 namespace miopen {
 
+struct AnyInvokeParams;
+
 namespace solver {
 struct ConvSolution;
```

Comment:
@DrizztDoUrden  any change we can make Algorithm name available in the Convolution Context? its needed to call handle.AddKernel() (which requires algorithm and network as first 2 params) in generic_search. 
I think NetworkConfig is available through context.BuildConfKey() but I dont see a way to access algorithm

Reply:
You may use `solver::Id{SolverDbId(s)}.GetAlgo()` for that, I suppose.
FYI, invokers don't use that layer of KernelDb, so you may also have to alter `Handle::PrepareInvoker` in some way.